src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -87,13 +89,18 @@ internal override ValueTask<HttpResponseMessage> SendAsync(HttpRequestMessage re
             }
         }
 
+        [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "EnableActivityTracker")]
+        private static extern void EnableActivityTracker(EventSource? instance);
+
         private async ValueTask<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
         {
             // HttpClientHandler is responsible to call static DiagnosticsHandler.IsEnabled() before forwarding request here.
             // It will check if propagation is on (because parent Activity exists or there is a listener) or off (forcibly disabled)
             // This code won't be called unless consumer unsubscribes from DiagnosticListener right after the check.
             // So some requests happening right after subscription starts might not be instrumented. Similarly,
             // when consumer unsubscribes, extra requests might be instrumented
+            if (!IsGloballyEnabled()) throw new InvalidOperationException("Diagnostics are not enabled.");
+            EnableActivityTracker((EventSource?)null);
```

Comment:
This seeems really odd as something that should be handled by `EventSource` for mono instead.

`DiagnosticsHandler` doesn't emit `EventSource` events, that's the job of `HttpTelemetry` in our case.

If I understand it correctly this is also automatically opting users into a TPL feature that they would previously have to opt-in for via `TasksFlowActivityIds`.
Why is that necessary, and if it is, should it be limited to Mono? Why is it not a concern in other places using `EventSource`?

Reply:
I just understood from @antonfirsov that distributed tracing IDs and TPL activity IDs are not the same concept. 😮 

I will remove this `EnableActivityTracker` thing

Reply:
Or at least remove it from DiagnosticsHandler and think about Mono TPL separately